src/python/UnifiedConfiguration.py

```diff
@@ -47,6 +49,12 @@ def appendToFirmware(firmware_file, product_name, lua_name, defines, config, lay
                 hardware = json.load(h)
                 if 'overlay' in config:
                     hardware.update(config['overlay'])
+                if rx_as_tx is not None:
+                    if hardware['serial_rx'] is None or hardware['serial_tx'] is None:
+                        sys.stderr.write(f'Cannot select this target as RX-as-TX\n')
+                        exit(1)
+                    if rx_as_tx == TXType.external and hardware['serial_rx']:
+                        hardware['serial_rx'] = hardware['serial_tx']
                 layout = (json.JSONEncoder().encode(hardware).encode() + (b'\0' * 2048))[0:2048]
```

Comment:
need to add the LED conversion as well right?
```py
if (!hardware['led_red'] && hardware['led']):
  hardware['led_red'] = hardware['led']
  hardware['led'] = # undefined?
```
We still need to fix this but RXes mostly define `led` but TXes use `led_red` for their status

Reply:
Ok, I'll do that. But we really do need to sort out the LED definitions properly at some time.

Reply:
One-flippin'-hundred percent. It has been on my todo list since I made my first receiver definition like 4 years ago. I feel like now at least I know exactly what the difference is though.